### DISCUSSION TITLE
Reduce File Size

### DISCUSSION THREAD

**Discussion by ggrigera** [QuestPDF/QuestPDF#327] [Q&A] [answered]
Hello everyone!
I am currently generating a new PDF report that I will leave attached so you can see it ([1001 - Indicadores FEB. 22-JUL. 22 (59).pdf](https://github.com/QuestPDF/QuestPDF/files/9394641/1001.-.Indicadores.FEB.22-JUL.22.59.pdf)). The reality is that previously, it was weighing 2Mb.
I thought that by installing the new version, the size would go down considerably, however, this did not happen since it went from 2Mb to 1.3Mb.

From what I was able to analyze, the chart that adds 1.1Mb to the report is the pie chart, however I have another chart in the report (the horizontal bar chart) that I am handling in the same way but it only weighs 15Kb.

Has it occurred to anyone that it could be what is being defined wrong so that its size is large?

Here's an example of the code I'm using to add the chart.

```
grid.Item()
	.PaddingTop(20)
	.PaddingLeft(-82)
	.PaddingRight((float)-0.5, Unit.Centimetre)
	.Width(270)
	.Border(0.5f)
	.BorderBottom(0)
	.BorderColor("#A5A5A5")
	.Height(180)
	.Background("#F8CBAD")
	.AlignCenter()
	.Element(element =>
	{
		element
			.PaddingTop(-8)
			.PaddingLeft(4)
			.AlignCenter()
			.Canvas((canvas, size) =>
			{
				string pieChartTemp = Path.GetTempFileName();
				using (var stremPieChart = File.Create(pieChartTemp))
				{
					var exporter = new OxyPlot.SvgExporter { Width = 400, Height = 300, UseVerticalTextAlignmentWorkaround = true };
					exporter.Export(pieChart, stremPieChart);
				}
				var pieChartSvg = new SKSvg();
				pieChartSvg.Load(pieChartTemp);
				canvas.Scale(0.99f);
				canvas.DrawPicture(pieChartSvg.Picture);

				File.Delete(pieChartTemp);
			});
	});
```

The library that I am using to generate the charts is OxyPlot, since MicrhoCharts did not allow me to generate this type of charts.

**Comment by bennetbo** [comment]
Does your chart have a lot of lines, points, etc.? Svg can get quite large if there are a lot of shapes inside. 
Try the [PNG exporter](https://oxyplot.readthedocs.io/en/latest/export/export-png.html) and check if that helps.

**Reply by ggrigera** [reply]
I saw that this option exists, but only for the platforms that it details, therefore, it does not serve me.
"The PNG export functionality is located in the platform specific libraries (OxyPlot.Wpf, OxyPlot.WindowsForms)"

**Reply by bennetbo** [reply]
Replied to quick i guess, you are correct. 
Checkout Marcins response though, as embedded font files (e.g. Calibri) tend to be quite large, because SkiaSharp performs no subsetting.

**Reply by ggrigera** [reply]
I added an answer with the solution that I was able to give you.

**Comment by MarcinZiabek** [comment]
Please check the font that is used in your chart. Best is to change the font to something smaller, e.g. `Lato`.

**Reply by ggrigera** [reply]
I added an answer with the solution that I was able to give you.

**Comment by ggrigera** [comment, marked as answer]
Finally, I solved it by directly inserting an image ("pieChart" parameter - FilePath) in the PDF since as I was doing it, I was generating it right there and "pieChart" model, it loaded all the fonts in the PDF.

Now the image is generated before starting to generate the PDF.

```
grid.Item()
	.PaddingTop(20)
	.PaddingLeft(-82)
	.PaddingRight((float)-0.5, Unit.Centimetre)
	.Width(270)
	.Border(0.5f)
	.BorderBottom(0)
	.BorderColor("#A5A5A5")
	.Height(180)
	.Background("#F8CBAD")
	.AlignCenter()
	.Element(element =>
	{
		element
			.PaddingTop(-8)
			.PaddingLeft(4)
			.AlignCenter()
			.Image(pieChart, ImageScaling.FitArea);
	});
```

Thanks for the help!!

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
